src/components/Table.tsx

```diff
@@ -81,42 +88,93 @@ const renderField = (row: TableDataRow, column: TableColumn) => {
 	return value == null ? '' : value;
 };
 
-export default class Table extends React.Component<
-	TableProps,
+export default class Table<T> extends React.Component<
+	TableProps<T>,
 	{
-		sortColumn: null | string;
+		allChecked: boolean;
 		reverse: boolean;
+		checkedItems: T[];
+		sortColumn: null | keyof T;
 	}
 > {
-	constructor(props: TableProps) {
+	constructor(props: TableProps<T>) {
 		super(props);
 
+		if (props.onChecked && !props.rowKey) {
+			throw new Error(
+				'A `rowKey` property must be provided if using `onChecked` with a Table component',
+			);
+		}
+
 		this.state = {
-			sortColumn: null,
+			allChecked: false,
 			reverse: false,
+			checkedItems: [],
+			sortColumn: null,
 		};
 	}
 
-	sortData(data: TableDataRow[]): TableDataRow[] {
+	isChecked(item: T) {
+		const rowKey = this.props.rowKey;
+		if (!rowKey) {
+			return false;
+		}
+
+		const identifier = item[rowKey];
+		return !!find(this.state.checkedItems, x => x[rowKey] === identifier);
```

Comment:
Could this be `some(this.state.checkedItems, { [rowKey]: identifier }`?

Reply:
Classy!!